pull_request_template.md

```diff
@@ -0,0 +1,27 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+
+# How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.
+
+
+
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
```

Comment:
I think we should clarify this, as I don't want to encourage noisy or unnecessary comments. Code should be self documenting where possible. Ideally, comments should talk about the why, the context that can't be gathered from clean, readable code, and in most cases never the what or how.

Perhaps something along the lines of

```suggestion
- [ ] I have made my code self-documenting where possible, with the use of clear variable and function names
- [ ] I have commented the reasons for any complex or unusual code, so others can understand the context
```